### ISSUE TITLE
(Feature Request): Return deleted comments

### ISSUE THREAD

**Issue by mPiotrek** [Anarios/return-youtube-dislike#261]
### Extension or Userscript?

Both

### Request or suggest a new feature!

In contrast to maintaining an alternative comments section (e.g. for videos with locked | disabled comments) this feature request is about preserving automatically deleted comments, so that the user does not lose them.

For some time now, YouTube has been removing certain comments.
Usually you can get them through if you tweak the wording a little or redact certain names and otherwise suspicious keywords. However, it requires you to copy the comment beforehand, refresh the comments a few times and usually wait two dozen or so seconds to see if the comment is still there.
It is not only annoying and a massive waste of time, but ultimately makes this censorship much more effective.

### Ways to implement this!

I am not sure if it is (or should be) in scope for [RYD](https://github.com/Anarios/return-youtube-dislike), or if it would be better served by a separate extension.

Now, to have it implemented the extension would need to:
* locally save every comment, just before posting
* some time after posting check if the comment is still up
* if not, notify the user and provide them with the saved copy

### Can you work on this?

I have no experience with browser extensions nor with userscripts.

**Comment by sy-b** [comment]
@mPiotrek 
Data Storage is a **huge** problem.

So, not possible.

**Comment by mPiotrek** [comment]
@sy-b
What do you mean? Can't extensions and userscripts access LocalStorage or equivalent?

**Comment by sy-b** [comment]
>In contrast to maintaining an alternative comments section (e.g. for videos with locked | disabled comments) this feature request is about preserving automatically deleted comments, so that the user does not lose them.

I interpreted that as "storing comments on server and browser" so that user can access it later
because, it can be easily erased on local storage. 

"Storing comments in browser storage" is possible but I don't think it would be reliable.
If user never cleans browser storage, then it will work fine.

I don't its (for now) out of scope of this extension.

**Comment by mPiotrek** [comment]
> "Storing comments in browser storage" is possible but I don't think it would be reliable. If user never cleans browser storage, then it will work fine.

I meant that it would be stored for a few minutes max. YouTube usually gets rid of 'unright' comments ~20 seconds after submitting them and only after such time would the user need to be notified. Retroactively removed comments are a separate issue tbh

> I think its out of scope of this extension (at least for now)

That's totally valid. Thanks for your prompt replies anyway.
I'm gonna leave the issue open a while longer, but if there's not gonna be anyone to pick up the idea I'll just close it and continue thinking on my own

**Comment by cupiditys** [comment]
cant really see this being in scope for RYD but possibly its own extension as filtered keywords added by the creator like "scam" or "virus" are a fairly easy way to tell if it's misleading

**Comment by mPiotrek** [comment]
Alright, 16 days is long enough so I'm closing this. In case of any updates I will post some comment here
